task_3/module/ec2/main.tf

```diff
@@ -0,0 +1,75 @@
+# DATASOURCES
+
+data "aws_ami" "ami" {
+  most_recent = var.ami_param.most_recent
+  name_regex  = var.ami_param.name_regex
+  owners      = var.ami_param.owners
+}
+
+data "aws_vpc" "vpc" {
+  id = var.vpc_id
+}
+
+data "aws_subnet" "subnet" {
+  vpc_id            = data.aws_vpc.vpc.id
+  availability_zone = var.availability_zone
+  filter {
+    name   = "tag:Name"
+    values = [var.subnet_name]
+  }
+}
+
+data "aws_caller_identity" "current" {}
+
+# ---------------------------------------
+# VARIABLES
+
+# ---------------------------------------
+# RESOURCES
+
+resource "aws_key_pair" "ssh_key" {
+  count      = var.instance_count > 0 ? 1 : 0
+  key_name   = var.ssh_public_key.name
+  public_key = var.ssh_public_key.value
+  tags = merge(var.tags, {
+    AWS_Account_ID = data.aws_caller_identity.current.account_id
+  })
+}
+
+resource "aws_instance" "instance" {
```

Comment:
Which VPC is this EC2 instance placed to?

Reply:
new resource property added
```terraform
resource "aws_instance" "instance" {
  subnet_id     = var.subnet_id
...
```